### PR TITLE
Experimental unique 3p iframe origin.

### PR DESCRIPTION
**Pull request by cramforce**
Origin changes per page view.

### CODE REVIEW

**Comment by dvoytenko** [comment]
LGTM

File: gulpfile.js [hunk 1]
```diff
@@ -403,7 +403,8 @@ function thirdPartyBootstrap(watch, shouldMinify) {
   console.log('Processing ' + input);
   var html = fs.readFileSync(input, 'utf8');
   var min = html;
-  min = min.replace(/\.\/integration\.js/g, './f.js');
+  min = min.replace(/\.\/integration\.js/g,
```
Comment (dvoytenko) [inline gulpfile.js]: This doesn't look right. Does it? You replacing a part of URL with the absolute URL?

File: src/3p-frame.js [hunk 2]
```diff
@@ -196,6 +198,30 @@ function getDefaultBootstrapBaseUrl(parentWindow) {
 }
 
 /**
+ * Sub domain on which the 3p iframe will be hosted.
+ * Because we only calculate the URL once per page, this function is only
+ * called once and hence all frames on a page use the same URL.
+ * @return {string}
+ */
+export function getSubDomain(win) {
```
Comment (dvoytenko) [inline src/3p-frame.js]: Is it exported for testing only? I'm a bit concerned that it's documented as a called-once, but there's no code to ensure it doesn't yield singular values. If it's private/testing-only - that's fine. But then i'd also name it as such.

File: gulpfile.js (hunk 1 above)
Comment (cramforce) [inline gulpfile.js]: I had to update this a bit for testing and added lots of comments.

This change will need to be tested in op-in prod. Should be a no-op, but changes the contents of the file. Previously we always used a relative URL.

File: src/3p-frame.js (hunk 2 above)
Comment (cramforce) [inline src/3p-frame.js]: Added comment. Its exported for testing only.

**Pull request merged**